### PR TITLE
Change get_patchelf to immediately return patchelf path if found

### PR DESCRIPTION
**Pull request by gartung**

### CODE REVIEW

**Comment by gartung** [comment]
No. Because I might want to put get_patchelf() back in binary_distribution later.

File: lib/spack/spack/relocate.py [hunk 1]
```diff
@@ -66,20 +66,20 @@ def get_patchelf():
...
-    if str(spack.architecture.platform()) == 'test':
-        return None
-    if str(spack.architecture.platform()) == 'darwin':
-        return None
     patchelf = spack.util.executable.which('patchelf')
-    if patchelf is None:
+    if patchelf is not None:
+        return patchelf.path
+    else:
+        if str(spack.architecture.platform()) == 'test':
```
[the remark this answers is not in the document]
Comment (gartung) [reply lib/spack/spack/relocate.py]: No. I might want to make use of get_patchelf() the get the rpaths in the creation of the buildinfo file in binary_distribution and this if statement is needed to allow the creation of the test buildcaches.

File: lib/spack/spack/relocate.py [hunk 2]
```diff
@@ -66,20 +66,20 @@ def get_patchelf():
...
+            return None
+        if str(spack.architecture.platform()) == 'darwin':
+            return None
         patchelf_spec = spack.cmd.parse_specs("patchelf", concretize=True)[0]
         patchelf = spack.repo.get(patchelf_spec)
         if not patchelf.installed:
             patchelf.do_install(use_cache=False)
```
Comment (tgamblin) [inline lib/spack/spack/relocate.py]: Is there a reason this won't use the cache?  We've had users ask why they always have to build patchelf even if it's in the cache

Comment (gartung) [reply lib/spack/spack/relocate.py]: Having patchelf patch itself seemed like a bad idea at the time.

Comment (gartung) [reply lib/spack/spack/relocate.py]: I better solution might be to download the executable but how do you guarantee libc compatibility.

Comment (gartung) [reply lib/spack/spack/relocate.py]: Having packages with only system libraries unpack without requiring patchelf would require extra metadata that is not recorded yet. Maybe a find like standalone=True in the buildinfo dictionary.

Comment (gartung) [reply lib/spack/spack/relocate.py]: @tgamblin I propose that I change this to an Exception if patchelf is not in $PATH and require that the user provide patchelf.

**Pull request merged**